cloud_utils/vault.py

```diff
@@ -1,136 +1,35 @@
 
+def _get_client(url: str, token: str) -> hvac.Client:
+    return hvac.Client(url=url, token=token)
 
-class KeyDoesNotExist(Exception):
-    pass
 
-
-async def _identity_token():
-    return (
-        (
-            await gamla.get_async_with_headers(
-                {"Metadata": "true"},
-                30,
-                f"{base_metadata_url}/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F",
-            )
-        )
-        .json()
-        .get("access_token")
-    )
-
-
-async def _instance_metadata():
-    return (
-        await gamla.get_async_with_headers(
-            {"Metadata": "true"},
-            30,
-            f"{base_metadata_url}/instance?api-version=2019-08-15",
-        )
-    ).json()
-
-
-def _make_pod_identity_token(
-    role: str,
-    base_vault_url: str,
-) -> Callable[[], Coroutine[Any, Any, str]]:
-    @AsyncTTL(time_to_live=60 * 60 * 23, maxsize=1)  # Vault token is valid for 24 hour.
-    async def _pod_identity_token() -> str:
-        jwt = await _identity_token()
-        return await gamla.pipe(
-            await _instance_metadata(),
-            gamla.get_in(["compute"]),
-            lambda compute: {
-                "role": role,
-                "jwt": jwt,
-                "subscription_id": gamla.get_in(["subscriptionId"])(compute),
-                "resource_group_name": gamla.get_in(["resourceGroupName"])(compute),
-                "vm_name": gamla.get_in(["name"])(compute),
-                "vmss_name": gamla.get_in(["vmScaleSetName"])(compute),
-            },
-            gamla.post_json_async(30, f"{base_vault_url}/auth/azure/login"),
-            lambda response: response.json(),
-            gamla.get_in(["auth", "client_token"]),
-        )
-
-    return _pod_identity_token
-
-
-def _make_vault_headers(
-    token: Callable[[], Coroutine[Any, Any, str]],
-) -> Callable[[], Coroutine[Any, Any, Dict[str, str]]]:
-    async def _vault_headers() -> Dict[str, str]:
-        return {
-            "X-Vault-Token": await token(),
-        }
-
-    return _vault_headers
+def read_all_secrets(path: str) -> dict[str, str]:
+    client = _get_client(_URL, _TOKEN)
+    read_secrets = client.secrets.kv.v2.read_secret_version(path=path)
+    return read_secrets["data"]["data"]
 
 
-def _make_read_key(
-    base_vault_url: str,
-    headers: Callable[[], Coroutine[Any, Any, dict]],
-):
-    async def read_key(path: str) -> Dict[str, str]:
-        return gamla.pipe(
-            await gamla.get_async_with_headers(
-                (await headers()),
-                30,
-                f"{base_vault_url}/secret/data/{path}",
-            ),
-            lambda response: response.json(),
-            gamla.excepts(
-                KeyError,
-                gamla.make_raise(KeyDoesNotExist),
-                gamla.get_in(["data", "data"]),
-            ),
-        )
+# TODO: do we want to have the ability to read a specific secret?
```

Comment:
Good question, what happens if you pass to read_all_secerts function a path that includes the key name? 
i.e. read_all_secerts("1234/prod/EDEN") 
Will this work? 
In general, I don't think we need a usecase of reading one specific key but rather a complete path (since the integration instance returns all secrets for that instance

Reply:
`read_all_secerts("1234/prod/EDEN")` won't work - 
from `read_all_secerts("1234/prod")` we get a dict in the form of - 
`{'request_id': ..., ....... , 'data': {'data': {'EDEN1': 'val1', '...}, 'metadata': {...}}, ...}`
in order to get the secrets we need to access `dict["data"]["data"]` - and that returns the dict `{'EDEN1': 'val1', '...}`